File: Ferret/Enums/MissionResult.cs
namespace Ferret.Enums;

public enum MissionResult
{
    Gold,
    Silver,
    Bronze,
}

public static class MissionResultExtensions
{
    public static string ToFriendlyString(this MissionResult self)
    {
        return self switch
        {
            MissionResult.Gold => "Gold",
            MissionResult.Silver => "Silver",
            MissionResult.Bronze => "Bronze",
            _ => "Gold",
        };
    }

    public static string ToLuaEnum(this MissionResult self)
    {
        return self switch
        {
            MissionResult.Gold => "MissionResult.Gold",
            MissionResult.Silver => "MissionResult.Silver",
            MissionResult.Bronze => "MissionResult.Bronze",
            _ => "MissionResult.Gold",
        };
    }
}

File: lib/Ferret/CosmicExploration/Data/MissionResult.lua
--------------------------------------------------------------------------------
--   DESCRIPTION: Mission result, fail, bronze, silver, gold etc.
--        AUTHOR: Faye (OhKannaDuh)
--------------------------------------------------------------------------------

---@alias MissionResult integer
MissionResult = {
    Fail = 1,
    Bronze = 2,
    Silver = 3,
    Gold = 4,
}

function MissionResult.to_string(value)
    local map = {
        [MissionResult.Fail] = 'Fail',
        [MissionResult.Bronze] = 'Bronze',
        [MissionResult.Silver] = 'Silver',
        [MissionResult.Gold] = 'Gold',
    }

    return map[value] or ('Unknown - ' .. tostring(value))
end
